src/testRunner/unittests/tsserver/exportMapCache.ts

```diff
@@ -129,6 +129,92 @@ describe("unittests:: tsserver:: exportMapCache", () => {
+        const classesTs: File = {
+            path: "/classes.ts",
+            content: `import { Component } from "./utils.js";
+
+            export class MyComponent extends Component {
+                render/**/
+            }`
+        };
+        const host = createServerHost([utilsTs, classesTs, tsconfig]);
+        const session = createSession(host);
+        openFilesForSession([classesTs], session);
+        session.executeCommandSeq<ts.server.protocol.ConfigureRequest>({
+            command: ts.server.protocol.CommandTypes.Configure,
+            arguments: {
+                preferences: {
+                    includeCompletionsForModuleExports: true,
+                    includeCompletionsWithClassMemberSnippets: true,
+                    includeCompletionsWithInsertText: true,
+                },
+            }
+        });
+        session.executeCommandSeq<ts.server.protocol.CompletionsRequest>({
+            command: ts.server.protocol.CommandTypes.CompletionInfo,
+            arguments: {
+                file: classesTs.path,
+                line: 4,
+                offset: 23,
+                prefix: "render",
+                includeExternalModuleExports: true,
+                includeInsertTextCompletions: true,
+            }
+        });
+
+        const projectService = session.getProjectService();
+        const project = configuredProjectAt(projectService, 0);
+        const exportMapCache = project.getCachedExportInfoMap();
+        assert.ok(exportMapCache.isUsableByFile(classesTs.path as ts.Path));
+        assert.ok(!exportMapCache.isEmpty());
+
+        openFilesForSession([{ file: utilsTs.path, content: utilsTs.content.replace("render", "render2") }], session);
+        session.executeCommandSeq<ts.server.protocol.UpdateOpenRequest>({
+            command: ts.server.protocol.CommandTypes.UpdateOpen,
+            arguments: {
+                changedFiles: [{
+                    fileName: classesTs.path,
+                    textChanges: [{
+                        newText: "",
+                        start: { line: 4, offset: 22 },
+                        end: { line: 4, offset: 23 },
+                    }]
+                }]
+            }
+        });
+
+        host.runQueuedTimeoutCallbacks();
+        project.getPackageJsonAutoImportProvider();
+
+        // Cache is invalidated because other file has changed
+        assert.ok(!exportMapCache.isUsableByFile(classesTs.path as ts.Path));
+        assert.ok(exportMapCache.isEmpty());
+
+        // Does not crash
+        session.executeCommandSeq<ts.server.protocol.CompletionsRequest>({
+            command: ts.server.protocol.CommandTypes.CompletionInfo,
+            arguments: {
+                file: classesTs.path,
+                line: 4,
+                offset: 22,
+                prefix: "rende",
+                includeExternalModuleExports: true,
+                includeInsertTextCompletions: true,
+            }
+        });

```

Comment:
Can you please baseline tsserver log so its easier to debug and find out whats going on when  esp some of the stuff about project creation/watches change,

Reply:
How do I do that?

Reply:
use createLoggerWithInMemoryLogs and baselineTsserverLogs eg: https://github.com/microsoft/TypeScript/blob/main/src/testRunner/unittests/tsserver/openFile.ts#L42